src/main/java/org/testng/TestRunner.java

```diff
@@ -544,9 +531,22 @@ private void initMethods() {
             true /* unique */,
             m_excludedMethods,
             comparator);
-    // shared group methods
-    m_groupMethods =
-        new ConfigurationGroupMethods(m_allTestMethods, beforeGroupMethods, afterGroupMethods);
+  }
+
+  private ITestNGMethod[] computeAndGetAllTestMethods() {
+    List<ITestNGMethod> testMethods = Lists.newArrayList();
```

Comment:
Proposal:

```java
if (m_allTestMethods != null && !RuntimeBehavior.isMemoryFriendlyMode()) {
  return m_allTestMethods;
}
```

Reply:
Fixed but we will refer to `m_allTestMethods` only when we are in `JUnit` mode.